src/org/openvv/OVVAsset.as

```diff
@@ -438,6 +429,46 @@ package org.openvv {
                 _intervalTimer.addEventListener(TimerEvent.TIMER, onIntervalCheck);
                 _intervalTimer.start();
             }
+    }
+
+        private function setStage(evt:Event = null):void
+        {
+            var ad:DisplayObject = _ad as DisplayObject;
+            if(!ad) return;
+
+            ad.removeEventListener(Event.ADDED_TO_STAGE, setStage);
+            try{
+                _stage = ad.stage;
+            }
+            catch(ignore:Error){
+                //stage is inaccessible
+            }
+            if(!_stage)
+                ad.addEventListener(Event.ADDED_TO_STAGE, setStage);
+        }
+
+        private function getDisplayState(results:OVVCheck):String
+        {
+            var hasStageAccess:Boolean;
+            var displayState:String = StageDisplayState.NORMAL;
+
+            try{
+                displayState   = _stage.displayState;
+                hasStageAccess = true;
+            }
+            catch(ignore:Error){
+                // Either stage was null or we can't access it due to security
+                // restrictions, either way we can ignore this error
+            }
+
+            if(!hasStageAccess && _ad && (_ad is DisplayObject))
+            {
+                if ((_ad.width - (results.objRight - results.objLeft)) > 10 && (_ad.height - (results.objBottom - results.objTop)) > 10) {
```

Comment:
Hmm, does everyone else agree with the assertion that "If ad is bigger than player, fullscreen is the only possible explanation?" Does this hold up when the browser is zoomed in or out? Perhaps it would be better to compare the ad's reported dimensions vs. JavaScript's screen.width, and screen.height properties?